pallets/inactivity-tracking/src/lib.rs

```diff
@@ -304,6 +315,40 @@ pub mod pallet {
             }
         }
 
+        /// Internal function to populate the current session active collator records with collators
+        /// part of inactive chains.
+        pub fn process_inactive_chains_for_session() {
+            match <CurrentActivityTrackingStatus<T>>::get() {
+                ActivityTrackingStatus::Disabled { .. } => return,
+                ActivityTrackingStatus::Enabled { start, end: _ } => {
+                    if start > T::CurrentSessionIndex::session_index() {
+                        return;
+                    }
+                }
+            }
+            let active_chains = <ActiveContainerChainsForCurrentSession<T>>::get();
+            let _ = <ActiveCollatorsForCurrentSession<T>>::try_mutate(
+                |active_collators| -> DispatchResult {
+                    T::CurrentCollatorsFetcher::container_chains_with_collators(
+                        ForSession::Current,
+                    )
+                    .iter()
+                    .for_each(|(para_id, collator_ids)| {
+                        if !active_chains.contains(para_id) {
+                            collator_ids.iter().for_each(|collator_id| {
+                                if !active_collators.contains(collator_id) {
+                                    let _ = active_collators
+                                        .try_insert(collator_id.clone())
+                                        .map_err(|_| Error::<T>::MaxCollatorsPerSessionReached);
```

Comment:
this ensures that  if the chain is inactive, we mark its collators as active?

Reply:
Yes!

Reply:
Updated the code to disable inactivity tracking if `ActiveCollatorsForCurrentSession` while processing inactive chains and added test for it in https://github.com/moondance-labs/tanssi/pull/899/commits/1ea2a0b3266a7ea66d31fd1d84a02ff3e9b92644 and https://github.com/moondance-labs/tanssi/pull/899/commits/428c36c8d4f8afeb8d2c9c53b341d3efb86ce357